hydromt/raster.py

```diff
@@ -1652,6 +1655,217 @@ def interpolate_na(self, method: str = "nearest", **kwargs):
         )
         return interp_array
 
+    def to_xyz(
+        self,
+        root: str,
+        px: int,
+        zoomlevels: list = [],
+        fmt: str = "tif",
+    ):
+        """Export rasterdataset to tiles in a xyz structure
+
+        Parameters
+        ----------
+        root : str
+            Path where the database will be saved
+            Database yml will be put one directory above
+        px : int
+            Amount of pixels per tile
+        zoomlevels : list, optional
+            Zoom levels to be put in the database
+
+        Raises
+        ------
+        ValueError
+            Amount of pixels can not be larger than the original dataset
+        """
+
+        mName = os.path.normpath(os.path.basename(root))
+
+        def folder(path):
+            if not os.path.exists(path):
+                os.makedirs(path)
+
+        folder(root)
+
+        mdim = min(self._obj.shape)
+        if px > mdim:
+            raise ValueError("")
+        nzl = int(np.ceil((np.log10(mdim / px) / np.log10(2))))
+        pxs = [px * 2**num for num in range(nzl)]
+
+        # Clean up zoomlevels list
+        zoomlevels = [x for x in zoomlevels if x <= len(pxs) - 1]
+
+        def tile_window(px):
+            # Basic stuff
+            nr, nc = self._obj.shape
+            lu = product(range(0, nc, px), range(0, nr, px))
+            ## create the window
+            for l, u in lu:
+                h = min(px, nr - u)
+                w = min(px, nc - l)
+                yield (l, u, w, h)
+
+        for zl in zoomlevels:
+            sd = f"{root}\\{zl}"
+            folder(sd)
+            # Write the raster paths to a text file
+            file = open(f"{sd}\\filelist.txt", "w")
+
+            for l, u, w, h in tile_window(pxs[zl]):
+                col = int(np.ceil(l / pxs[zl]))
+                row = int(np.ceil(u / pxs[zl]))
+                ssd = "{}\\{}".format(sd, col)
+
+                folder(ssd)
+
+                # create temp tile
+                temp = self._obj[u : u + h, l : l + w].load()
```

Comment:
We could avoid this load method if we start from the previous zoom level at each iteration.

Reply:
done